### PR TITLE
app-doc/doxygen: version bump to 1.8.12 and move to EAPI 6

### PR DESCRIPTION
**Pull request by bjacquin**
Move to EAPI 6
    Replace dodoc with DOCS
    Replace epatch with PATCHES
    Drop python 3.3 support

### CODE REVIEW

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+pkg_setup() {
+	use doc && python-any-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+
+	# Ensure we link to -liconv
+	if use elibc_FreeBSD && has_version dev-libs/libiconv || use elibc_uclibc; then
+		for pro in */*.pro.in */*/*.pro.in; do
```
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: declare `pro` local before

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+		echo "unix:LIBS += -liconv" >> "${pro}"
```
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: missing `|| die`

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		elog "Continuing with -O2 instead ..."
+		echo
+		replace-flags "-O3" "-O2"
+	fi
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DDOC_INSTALL_DIR="share/doc/${P}"
+		$(cmake-utils_use clang use_libclang)
```
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: you sure these don't cause problems in `EAPI=6`?

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+	# generate html and pdf documents. errors here are not considered
+	# fatal, hence the ewarn message.
+
+	if use doc; then
+		export VARTEXFONTS="${T}/fonts" # bug #564944
+
+		if ! use dot; then
+			sed -i -e "s/HAVE_DOT               = YES/HAVE_DOT    = NO/" \
+				{Doxyfile,doc/Doxyfile} \
+				|| ewarn "disabling dot failed"
```
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: this needs to be `die`, not `ewarn`

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+		fi
+		cd "${BUILD_DIR}" && emake docs
```
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: split this into:
```
cd "${BUILD_DIR}" || die
emake docs
```

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+	fi
+}
+
+src_install() {
+	default
```
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: if you use cmake-utils.eclass, you don't use `default`. Put the `cmake-utils_src_install` here instead.

**Comment by bjacquin** [comment]
Indeed, all is fixed in last PR. Thanks for the review

**Comment by tamiko** [comment]
@bjacquin Thank you very much for the patches.

Shall I add you to the maintainers list such that you get CCed on bugs?
(Next time just ping me on PRs directly.)

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+src_install() {
+	if use qt5; then
+		doicon "${DISTDIR}/doxywizard.png"
+		make_desktop_entry doxywizard "DoxyWizard ${PV}" \
+			"/usr/share/pixmaps/doxywizard.png" \
+			"Development"
+	fi
+
+	cmake-utils_src_install
+	einstalldocs
```
Comment (doughdemon) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: ``cmake-utils_src_install`` already calls ``einstalldocs``

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+	qt5? (
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5
+	)
+	sqlite? ( dev-db/sqlite:3 )
+	"
+
+REQUIRED_USE="doc? ( latex )"
+
+DEPEND="sys-apps/sed
```
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: you sure you need sed? I thought it was part of base, please check this...

File: app-doc/doxygen/doxygen-1.8.12.ebuild [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+src_prepare() {
+	default
+
+	# Ensure we link to -liconv
+	if use elibc_FreeBSD && has_version dev-libs/libiconv || use elibc_uclibc; then
+		local pro
+		for pro in */*.pro.in */*/*.pro.in; do
+		echo "unix:LIBS += -liconv" >> "${pro}" || die
```
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: indent this echo

File: app-doc/doxygen/doxygen-1.8.12.ebuild (hunk 7 above)
Comment (SoapGentoo) [inline app-doc/doxygen/doxygen-1.8.12.ebuild]: yes, nuke this `einstalldocs`, and prefer to put `cmake-utils_src_install` at the beginning of the phase function

**Comment by bjacquin** [comment]
Thanks for the comments, fixed in last PR

**Comment by tamiko** [comment]
Thanks a lot. Merged.

**Pull request closed without merging**